src/vis_cpu/vis_cpu.py

```diff
@@ -194,6 +194,15 @@ def vis_cpu(
             0 <= i < nbeam for i in beam_idx
         ), "beam_idx contains indices greater than the number of beams"
 
+    if beam_list is not None:
+        # make sure we interpolate to the right frequency first.
+        beam_list = [
+            b.interp(freq_array=np.array([freq]), new_object=True, run_check=False)
+            if (isinstance(freq, UVBeam) and freq not in b.freq_array)
```

Comment:
I think you mean `isinstance(b, UVBeam)` here.

Reply:
Yup.